design-proposals/vm-pool.md

```diff
@@ -0,0 +1,194 @@
+# Overview
+
+This design provides an approach for creating a VM grouping and replication abstraction for KubeVirt called a VirtualMachinePool. 
+
+## Motivation
+
+The ability to manage groups (or fleets) of similar VMs using a higher level abstraction is a staple among commonly utilized Iaas operational patterns. By bringing stateful VM group management to KubeVirt, we open the door for operation teams to utilize their existing patterns for managing KubeVirt VMs. This feature further aligns KubeVirt as an Iaas offering comparable to the public clouds which makes KubeVirt a more attractive option for Iaas management on baremetal hardware.
+
+## Goals
+
+* VM pool abstraction capable of managing replication of stateful VMs at scale
+* Automated rollout of spec changes across and other updates across a pool of stateful VMs
+* Automated and manual scale-out and scale-in VMs associated with a pool
+* Autohealing (delete and replace) of VMs in a pool not passing health checks
+* Ability to specify unique secrets and configMap data per VM within a VMPool
+
+## Non-Goals
```

Comment:
can you provide some more details about what you mean by `controlled scale-out`?

Reply:
What I mean by controlled scale-out is having a policy that defines how scale-out happens e.g. some form of ordering.  You defined controlled scale-in in `DisruptionStrategy`, so this would be the inverse.